scipy/stats/_distn_infrastructure.py

```diff
@@ -860,7 +866,10 @@ def freeze(self, *args, **kwds):
             The frozen distribution.
 
         """
-        return rv_frozen(self, *args, **kwds)
+        if hasattr(self, 'pdf'):
+            return rv_continuous_frozen(self, *args, **kwds)
```

Comment:
This _might_ fail:

```python
from scipy import stats

stats.binom.pdf = "my attribute"
frozen_binom = stats.binom()  # oops, returns rv_continuous frozen
```

```suggestion
        if isinstance(self, rv_continuous):
            return rv_continuous_frozen(self, *args, **kwds)
```

Reply:
Also, I think it'd be good to add this as a test.

Reply:
I avoided that because I've had trouble with `isinstance` checks with distributions. Sometimes they work and sometimes not, and I haven't thought carefully about why. But I can try it.

Reply:
When do these checks fail? As this is called by the ``__call__`` method of the distribution class itself, I don't think there should be any cases where this fails, right?

Reply:
I can't imagine when or why it happened in the past. 
I think it's fine. I'll push shortly.

Reply:
While I've got you, @tirthasheshpatel, do you have an opinion about gh-15632?

Reply:
I will get to it today. A lot to do in little time :)

Reply:
Of course. 
OK, I pushed the suggested change here.